app/controllers/bike_share_app.rb

```diff
@@ -29,17 +30,19 @@ class BikeShareApp < Sinatra::Base
   end
 
   get "/stations/:id/edit" do
-    ## if you want to have the user select city, installation date, and/or block count with a drop down, add instance variables like the one below.
-    ## @cities = City.all.order(:name)
     @station = Station.find(params[:id])
     erb :"stations/edit"
   end
 
   put "/stations/:id" do
-    ## if you want to have the user select city, installation date, and/or block count with a text field,
-    ##     you must create a new object or find an existing object with the data the user entered and then replace that text with the id in the parameters to create the station
-    ##     params[:city] = City.find_or_create_by(name: params[:city]).id
-    @station = Station.update(params[:id],params[:station])
+    Station.update(name: params[:station][:name],
+     city_id: City.find_or_create_by(params[:city]).id,
+     dock_count_id: DockCount.find_or_create_by(params[:dock_count]).id,
```

Comment:
What all is on the dock_count table? If you think of your Station table, it looks to me like you're going to have numbers here in this table that are just referring to numbers in another table. I can't think of any performance advantages, and it will likely actually make other queries more complicated.

Recommend storing the number of docks directly here in the Station table.